publish.go

```diff
@@ -145,26 +146,38 @@ func NewPublisher(url string, config amqp.Config, optionFuncs ...func(*Publisher
 
 	chManager, err := newChannelManager(url, config, options.Logger)
 	if err != nil {
-		return Publisher{}, nil, err
+		return Publisher{}, err
 	}
 
 	publisher := Publisher{
 		chManager:                  chManager,
-		notifyReturnChan:           make(chan Return, 1),
 		disablePublishDueToFlow:    false,
 		disablePublishDueToFlowMux: &sync.RWMutex{},
 		logger:                     options.Logger,
+		notifyReturnChan:           make(chan Return),
```

Comment:
We should probably close this channel at some point as well, so consumers can easily `range` over it. Maybe after the `notifyCancelOrClose` has been closed, around line `170`? A little tricky since we should safeguard (or think/make sure) against the `startNotifyReturnHandler` having also finished sending to the channel, otherwise it could panic if we `close` before it's done.

Reply:
Leaving a note for myself - It's actually when we call `StopPublishing` that this kind of cleanup needs to be triggered